Update `Select` docstring

dwierichs opened the pull request:
**Context:**
The `Select` operation has many names and symbols in the literature.

**Description of the Change:**

Changes to `Select` docstring:
1. Update the circuit drawing in the docstring (see below for convenience) to introduce box-control and mixed-circle-control nodes.
2. Add a brief mention of different names for `Select` operation in the literature: "Multiplexer", "multiplexed operation"
3. Add a mention of additional name if target operations are all single-qubit Pauli rotations about the same axis ("uniformly controlled rotation")
4. Add seealso link to `SelectPauliRot`, which is a uniformly controlled rotation.

Changes to `SelectPauliRot` docstring:
1. Make the naming priority conform with that of `Select` docstring: "Select" is the generic name (as given by the class name), "multiplexer/multiplexing" and "uniformly controlled rotation" are secondary, alternative names.
2. Mention rotation axis in equation and text. Mention that it is the same axis for all Select-applied rotations.
3. Add image like the one for `Select`.

**Benefits:**
Better coverage of naming and drawing conventions in literature.
More homogeneous docs between `Select` and `SelectPauliRot`.

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
[sc-91687]

Qottmann commented on doc/_static/templates/subroutines/select.tex:
oh neat I didnt see this! great idea!

Qottmann reviewed (approved):
awesome, thanks @dwierichs !

For some reason the RTD preview doesnt have links on the qml overiew, but the direct URL works and looks good :) https://xanaduai-pennylane--7464.com.readthedocs.build/en/7464/code/api/pennylane.Select.html

Qottmann commented on doc/_static/templates/subroutines/select.tex:
```diff
@@ -1,15 +1,26 @@
-\usepackage{quantikz}
+\usepackage{amsmath,quantikz}
 
+\tikzset{
+    selectctrl/.style={
+        draw,
+        circle,
+        minimum width=3.5pt,
+        inner sep=2.5pt,
+        append after command={
+            \pgfextra {
+                \filldraw[fill=white, color=white] (0, 0)-- (45:3.5pt) arc (45:225:3.5pt) -- cycle;
+                \fill (0, 0)-- (45:3.5pt) arc (45:-135:3.5pt) -- cycle;
+            }
+        }
+    }
+}
```
🧠 🤯

KetpuntoG reviewed (commented):
Nice! Thanks for the update 😄 Just a small comment,
Could you change the thumbnail of Select for a crop version of the current one?
It look very small in the website.
The path to the thumbnail image is added in this file: https://github.com/PennyLaneAI/pennylane/blob/master/doc/introduction/templates.rst
And don't forget to update the changelog (I would add it in the documentation section https://github.com/PennyLaneAI/pennylane/blob/master/doc/releases/changelog-dev.md#documentation-) 

otherwise, look good to go :)

dwierichs commented:
Very good pointer to both the thumbnails and the changelog, thanks @KetpuntoG !
- I cropped the image for Select and added it as second image.
- I did the same for SelectPauliRot, in order to make it more consistent, notation-wise, and I adapted the file location :) 
- Added changelog entry.

KetpuntoG commented:
![Screenshot 2025-05-23 at 10 08 41 AM](https://github.com/user-attachments/assets/acfbbd19-f711-473d-b5e8-c3e82969f68f)

It looks like you did it correctly but the image did not load. Can you confirm that it works for you?

dwierichs commented:
Seems fine for me :thinking: 
![image](https://github.com/user-attachments/assets/a5574655-1c79-4b4a-b373-ded6310d10fe)

KetpuntoG reviewed (approved):
Working now on my side too 💪

DSGuala commented on pennylane/templates/subroutines/select.py:
```diff
@@ -25,26 +25,34 @@
 class Select(Operation):
-    r"""Applies specific input operations depending on the state of
-    the designated control qubits.
+    r"""Applies individual operations depending on the state of
```
Is this easier/plainer language? (not a required change)
```suggestion
    r"""Applies different operations depending on the state of
```

DSGuala commented on pennylane/templates/subroutines/select.py:
```diff
@@ -25,26 +25,34 @@
+                    :width: 70%
                     :target: javascript:void(0);
 
+    This operator also is known as **multiplexer**, or multiplexed operation.
```
```suggestion
    This operator is also known as **multiplexer**, or multiplexed operation.
```

DSGuala commented on pennylane/templates/subroutines/select.py:
```diff
@@ -25,26 +25,34 @@
                     :target: javascript:void(0);
 
+    This operator also is known as **multiplexer**, or multiplexed operation.
+    If the applied operations :math:`\{U_i\}` all are single-qubit Pauli rotations about the
```
```suggestion
    If the applied operations :math:`\{U_i\}` are all single-qubit Pauli rotations about the
```

DSGuala commented on pennylane/templates/subroutines/select.py:
```diff
@@ -25,26 +25,34 @@
 
+    This operator also is known as **multiplexer**, or multiplexed operation.
+    If the applied operations :math:`\{U_i\}` all are single-qubit Pauli rotations about the
+    same axis, with the angle determined by the control qubits, this is also called a
+    **uniformly controlled rotation** gate.
```
Nice :+1:

DSGuala commented on pennylane/templates/subroutines/select_pauli_rot.py:
```diff
@@ -21,19 +21,26 @@
 class SelectPauliRot(Operation):
-    r"""Applies the multiplexer with Pauli rotations.
+    r"""Applies individual single-qubit Pauli rotations depending on the state of
+    designated control qubits.
```
I like this more, but up to you :slightly_smiling_face: 
```suggestion
    the designated control qubits.
```

dwierichs replied:
How about "some"? :) I wanted to skip the "the" because this is the beginning of a text. On the other hand, the signature of the function will appear above it in the docs... 
I don't know, what do you think about "some" :D

DSGuala replied:
Happy to keep it as is then!

DSGuala commented on pennylane/templates/subroutines/select.py:
```diff
@@ -25,26 +25,34 @@
-    r"""Applies specific input operations depending on the state of
-    the designated control qubits.
+    r"""Applies individual operations depending on the state of
+    designated control qubits.
```
I like this more but up to you :slightly_smiling_face: 
```suggestion
    the designated control qubits.
```

DSGuala commented on pennylane/templates/subroutines/select_pauli_rot.py:
```diff
@@ -21,19 +21,26 @@
 
-    This operator, also called a "multiplexed rotation", applies a sequence of uniformly controlled
-    rotations to a target qubit. The rotations are selected based on the values encoded in the control qubits.
+    This operator, also called a **multiplexed rotation** or **uniformly controlled rotation**,
```
Any reasons against adding aliases `qml.Multiplexer` and `qml.UniformlyControlledRotation`? We could benefit from users landing on our docs when they google these operations.

dwierichs replied:
`qml.Multiplexer` should be an alias for `qml.Select`.
`qml.UniformlyControlledRotation` or `qml.MultiplexedRotation` could be an alias for `SelectPauliRot` :) 

cc @KetpuntoG @josh146, who will have opinions, I'd anticipate. Also maybe cc @co9olguy ? :)

co9olguy replied:
these gates go by so many names that aliases are probably a good call (assuming they are ones we've seen used and we're not just making up new ones)

dwierichs replied:
see #7585

DSGuala reviewed (commented):
Nice improvement! Left some small suggestions, happy to approve once they're addressed or dismissed :slightly_smiling_face:
